18F/identity-idp
# Finalize cleanup for MFA selection presenters

## 🎫 Ticket

Follows the work of [LG-10687](https://cm-jira.usa.gov/browse/LG-10687), [LG-11145](https://cm-jira.usa.gov/browse/LG-11145), [LG-11147](https://cm-jira.usa.gov/browse/LG-11147), [LG-11140](https://cm-jira.usa.gov/browse/LG-11140), [LG-11141](https://cm-jira.usa.gov/browse/LG-11141), [LG-11144](https://cm-jira.usa.gov/browse/LG-11144)

## 🛠 Summary of changes

Changes include cleanup following the last of the refactored selection presenters, notably:

- Fixing a few occurrences of classes mistakenly not extending the new base classes (cb87669703bec85c6514bf59331060e49d001b72)
- Removing the now-unused previous base class (e8b6dcf1ab56fab5bf210b26864343a138e64fae)
- Consolidating `method` and `type` methods to a single `type` method, due to simplicity, some feedback at https://github.com/18F/identity-idp/pull/9560#discussion_r1396447478, and the fact that we only had test coverage for the `type` method anyways.
   - [Previously](https://github.com/18F/identity-idp/pull/9560#discussion_r1397302026), I mentioned a desire to get rid of this method altogether, but it's still used in [ID generation for list items in MFA selection lists](https://github.com/18F/identity-idp/blob/515a9197a6ebfeea31c63ef68bd4aad0cb3cb479/app/views/partials/multi_factor_authentication/_mfa_selection.html.erb#L8).
   - A few sub-refactorings:
      - Incorporate a [suggested rename](https://github.com/18F/identity-idp/pull/9560#discussion_r1396447478) of the phone-specific handling of delivery method (b13da097e4ba6f81503d9711536618dfe07ae6d2)
      - Change base class to raise `NotImplementedError` for expected implementation of subclasses (10ec6837a5d6826365cb5bc6e8e8bf91d1b888ce). This was more relevant when I was originally targeting to remove `type` altogether, but still an improvement IMO.

This ended up being a bit more extensive than I'd originally planned, so I'm happy to split this up across multiple pull requests if it's easier for review, especially since the commits are already fairly well-isolated.

## 📜 Testing Plan

Test both account creation flows and sign-in flows for MFA selection.

1. Go to http://localhost:3000
2. Create an account
3. Complete account creation up to MFA selection
4. Observe no regressions in the appearance of MFA selection
5. Continue account creation
6. On account dashboard, click "Forget all browsers" and confirm
7. Sign out
8. Sign in
9. When prompted for MFA, click "Choose another authentication method" troubleshooting option
10. Observe no regressions in the appearance of MFA selection
